### DISCUSSION TITLE
TwoTimesExecuteEffect nie działa w połączeniu z RandomizeDeckEffect

### DISCUSSION THREAD

**Discussion by itchystone** [emkarcinos/WMIAdventure#657] [Codding]
`TwoTimesExecuteEffect` nie działa jak trzeba, jeśli połączymy go z `RandomizeDeckEffect`

Pokaże, dlaczego:

### 1. Początek tury, `cards_queue` i `temp_cards_queue`:
![image](https://user-images.githubusercontent.com/43492764/141357834-11508b1d-0312-4074-ba44-1e154ed290cc.png)
![image](https://user-images.githubusercontent.com/43492764/141357880-b24f2c5c-d007-4bb7-9aa5-d7e8e64226f1.png)

### 2. Użycie `TwoTimesExecuteEffect`
![image](https://user-images.githubusercontent.com/43492764/141358277-a8945833-21f1-483f-a438-b0b73d378154.png)
![image](https://user-images.githubusercontent.com/43492764/141358111-6a1ec603-6db0-404a-87dd-a39fb7ad88c2.png)

### 3. Użycie `RandomizeDeckEffect`

![image](https://user-images.githubusercontent.com/43492764/141368878-e336744f-34c8-4e18-ba2d-ab9ab4316405.png)
![image](https://user-images.githubusercontent.com/43492764/141367925-10ae363c-0150-4f92-b7e9-0d614746f494.png)
![image](https://user-images.githubusercontent.com/43492764/141367938-f14b44f7-04a7-4451-bcba-12f560de071a.png)

#### Scenariusz 1: niepoprawny indeks (5) nie jest na początku `new_cards_order`
![image](https://user-images.githubusercontent.com/43492764/141368789-3245e417-86e9-4c67-b616-2df8492d7d64.png)

#### Scenariusz 2: niepoprawny indeks (5) jest na początku `new_cards_order`
![image](https://user-images.githubusercontent.com/43492764/141370784-fe223833-6a33-45b9-800e-91fbf377f06a.png)
![image](https://user-images.githubusercontent.com/43492764/141371061-90e15d69-627a-4ef4-8119-0f17dc865fbf.png)

**Comment by emkarcinos** [comment]
O fiuuu, to nie będzie izi do naprawienia

**Reply by itchystone** [reply]
chyba mam pomysł, ale zobacze czy ma sens jak zaczne go robić

**Reply by mattyl006** [reply]
o shiet

**Comment by itchystone** [comment]
a ogólnie to jak to powinno prawidłowo działać, tak jak w 1, czy 2?

![image](https://user-images.githubusercontent.com/43492764/141373711-1001ef52-ef03-464d-8ced-d1db1edb9374.png)

**Reply by emkarcinos** [reply]
Pamiętam że kiedyś mieliśmy taką dyskusję i wyszło, że najlepiej zrobić to tak zeby było najłatwiej dla nas, bo to nie ma różnicy

**Reply by mattyl006** [reply]
Czyli po każdej losowej zmianie w przypadku 2. będzie w kolejce jedna karta więcej?

**Reply by emkarcinos** [reply]
Tak, ale to tylko po stronie backendu, frontend dostanie to jako buff czy coś. W tym wypadku ta wersja 1. bardziej pasuje

**Reply by itchystone** [reply]
> 
> 
> Czyli po każdej losowej zmianie w przypadku 2. będzie w kolejce jedna karta więcej?


W sumie nawet nie po losowej zmianie, tylko po już użyciu `TwoTimesEffect`

(Tak to powinienem narysować)
![image](https://user-images.githubusercontent.com/43492764/141374768-86b98d32-f9ed-455a-b83d-8c03729712f0.png)

**Reply by itchystone** [reply]
też jestem za 1 wersją

**Reply by itchystone** [reply]
bo mamy design na 5 kart

**Reply by mattyl006** [reply]
2. wydaje mi się dzika jakaś, że może później problemy sprawić

**Comment by itchystone** [comment]
A jak zduplikujemy zablokowaną kartę, to duplikat też jest zablokowany?

**Reply by mattyl006** [reply]
Tam chyba chodzi z tą duplikacją, że wykonaj kartę dwa razy. To tak na logikę, gdy zablokujesz, to ten duplikat też powinien być zablokowany. Ale jeśli to bardzo komplikuje, to można zrobić że te drugie wykonanie nie jest zablokowane

**Reply by emkarcinos** [reply]
Wydaje mi się że powinniśmy pozbyć się tej mechaniki przetrzymywania karty która jest zduplikowana jako dwie karty i potratkować to jako buff

**Reply by itchystone** [reply]
okay
